### DISCUSSION TITLE
[Application] DScan

### DISCUSSION THREAD

**Discussion by akhileshthite** [filecoin-project/community#410] [Project Showcase]
![thumbnail](https://user-images.githubusercontent.com/68826419/150107083-43f29353-738a-4d9f-a231-7d4abb5c00fe.jpeg)

A decentralized storage and file-sharing tool that uploads content to IPFS and generates decentralized QR codes.

GitHub: https://github.com/p2plabsxyz/dscan
Extension: https://chrome.google.com/webstore/detail/dscan-decentralized-qr-co/idpfgkgogjjgklefnkjdpghkifbjenap

*Let us know if you have any questions down below.*

**Comment by akhileshthite** [comment]
### [v1.1](https://github.com/AkhileshThite/dscan/releases/tag/v1.1)

1. UI improvements
2. Security updates - Fix to prevent (XSS) attacks.

**Comment by akhileshthite** [comment]
### [v1.2](https://github.com/AkhileshThite/dscan/releases/tag/v1.2)

1. Fix for copy CID issue
2. Button to download generated QR code

**Comment by akhileshthite** [comment]
### [v1.3](https://github.com/buidltools/dscan/tree/v1.3) (webpack upgrade)
1. Webpack upgrade
-- Performance improvement (using dependencies over CDN packages. External packages take time to load the code as they are hosted on the cloud).
-- Advanced Webpack security.
2. Infura API for permanent pinning of the content
3. UI improvements
-- Loader animation.
-- Interactive UI.

**Comment by akhileshthite** [comment]
Hey folks!
I just created the "BUIDL Tools" organisation.
DScan is now under [BUIDL Tools](https://github.com/buidltools), we're building curated web3 infrastructure tools for the decentralized internet.

**Comment by akhileshthite** [comment]
### [v2.0](https://github.com/buidltools/dscan/releases/tag/v2.0) (web3.storage integration) 🚀

1. Web3.storage integration
2. Directory upload support
3. Fast upload speed (will take around half hour to upload upto 1GB of data)
4. Manifest version3
-- Removes all the external CDN packages.
5. Major UI improvements


![Screenshot 2022-03-08 at 8 54 52 PM](https://user-images.githubusercontent.com/68826419/157270635-4c64f4bc-8989-4a16-9aee-b6f55e185509.png)

**Comment by akhileshthite** [comment]
YouTube video: https://www.youtube.com/watch?v=Cl9I4wVhuSU

**Reply by web3jenks** [reply]
Love it!

**Reply by akhileshthite** [reply]
> Love it! 

Thank you :)

**Comment by akhileshthite** [comment]
### [v2.1](https://github.com/buidltools/dscan/tree/v2.1)
1. "Copied" popup message after copying IPFS CID.
2. Fix for "file" and "folder" button UI issue.

**Comment by akhileshthite** [comment]
### [v2.2](https://github.com/buidltools/dscan/tree/v2.2)
1. Fixes [issue 16](https://github.com/buidltools/dscan/issues/16)
2. UI improvements

**Comment by akhileshthite** [comment]
### [v2.3](https://github.com/buidltools/dscan/tree/v2.3)
1. Generate high resolution qr code image for downloading.
2. Fix for Infura public gateway is not supported anymore.

**Comment by akhileshthite** [comment]
### [v3.0](https://github.com/buidltools/dscan/tree/v3.0)
1. Show percentage progress during upload
2. Refactor {wrapWithDirectory: false}
3. Update packages
4. Improve Readme

**Comment by akhileshthite** [comment]
### [v3.1](https://github.com/buidltools/dscan/tree/v3.1)
1. QR code colour customization
2. Fix for https://github.com/buidltools/dscan/pull/34
3. Gateway update for faster performance (w3s.link)
4. UI improvements

![demo](https://user-images.githubusercontent.com/68826419/197556502-2bc3613a-7d72-481c-b3c7-5fab135eaf01.gif)

**Comment by akhileshthite** [comment]
### [v3.2](https://github.com/p2plabsxyz/dscan/tree/v3.2)
1. Store user's web3.storage API keys in local storage
2. Update organization name

![demo](https://user-images.githubusercontent.com/68826419/227793412-ee6dcc3f-641a-4f89-ade1-9f4a3286d010.gif)
![dscan_homepage](https://user-images.githubusercontent.com/68826419/227793636-f387008e-fbc0-4013-bc46-c0dc776ad144.png)

**Comment by akhileshthite** [comment]
### [v3.3](https://github.com/p2plabsxyz/dscan/tree/v3.3)
* Package updates
* URI for file uploads (wrapWithDirectory: true)
* Preserve directory structure when uploading folders
![v3 3](https://github.com/p2plabsxyz/dscan/assets/68826419/ef8a36bd-e7b3-4fe6-b189-a9cf64447728)

**Comment by akhileshthite** [comment]
### [v4.0](https://github.com/p2plabsxyz/dscan/tree/v4.0)
- w3up-client integration
  - Self-sovereign identity
  - User-owned storage
- Customizable QR code styles and logo integration

Learn more about w3up in this [blog](https://blog.web3.storage/posts/the-data-layer-is-here-with-the-new-web3-storage)

![demo](https://github.com/filecoin-project/community/assets/68826419/dc46f4b2-f467-4266-bc8e-c634f7f56843)
